tests/Gaufrette/Functional/Adapter/AwsS3Test.php

```diff
@@ -188,4 +188,16 @@ public function shouldUploadWithGivenContentType()
 
         $this->assertEquals('text/html', $this->filesystem->mimeType('foo'));
     }
+
+    /**
+     * @test
+     */
+    public function shouldRenameAnObject()
+    {
+        $this->filesystem->write('foo', '');
+        $this->filesystem->rename('foo', 'foo%_encode');
+
+        $this->assertTrue($this->filesystem->has('foo'));
```

Comment:
shouldn't the original name not be present ?

Reply:
You are right 👌 
Anyway there is no environment where this tests for S3 are executed?